### ISSUE TITLE
T.38 passthrough failure

### ISSUE THREAD

**Issue by jpyle490** [sipwise/rtpengine#1201]
This is on version 9.2.1.2 with OpenSIPS 3.1.

Alice and Bob are on a call through OpenSIPS with rtpengine handling the media.  rtpengine was engaged with:

```rtpengine_offer("ICE=remove DTLS=disable address-family=IP4 replace-origin replace-session-connection via-branch=extra");```
  and
```rtpengine_answer("ICE=remove DTLS=disable replace-origin replace-session-connection via-branch=extra");```

Bob sends a T.38 re-invite.  His SDP looks like this:

```
v=0
o=name 1614699264 1614699267 IN IP4 192.168.171.230
s=name
c=IN IP4 192.168.171.230
t=0 0
m=image 20334 udptl t38
a=T38FaxVersion:0
a=T38MaxBitRate:14400
a=T38FaxFillBitRemoval
a=T38FaxRateManagement:transferredTCF
a=T38FaxMaxBuffer:2000
a=T38FaxMaxDatagram:400
a=T38FaxUdpEC:t38UDPRedundancy
```

OpenSIPS treats it with
```rtpengine_offer("ICE=remove DTLS=disable force-answer replace-origin replace-session-connection");```
on its sequential request route, and rtpengine generates the following log entry:
```[core] Failed to create T.38 gateway: Failed to init PCM codec```

It prepares this SDP for Alice:
```
v=0
o=name 1614699264 1614699267 IN IP4 100.100.100.100
s=name
c=IN IP4 100.100.100.100
t=0 0
m=image 16840 RTP/AVP t38
a=sendrecv
a=rtcp:16841
a=ptime:20
```

Alice does her best with that, and replies with something that kind of makes sense, but rtpengine doesn't like it and complains of an ```Incomplete SDP specification```.  I don't blame it.  

What's happening on T.38 re-invite to cause the ugly SDP?  rtpengine seems like it's trying to set up a pcm/t38 gateway, which isn't what I'm looking for.  Did I configure something wrong or is this a bug?

**Comment by rfuchs** [comment]
Post a full debug log, otherwise I can't answer that.

**Comment by jpyle490** [comment]
Certainly.  Since it contains live public addressing, can I send it to you privately?

**Comment by rfuchs** [comment]
Yes of course. Email is in my profile.

**Comment by rfuchs** [comment]
Ok, so what happens here is that the T.38 re-invite is passed as an `answer` to rtpengine instead of an `offer`, which is wrong. So rtpengine sees a T.38 answer in response to a non-T.38 offer, which confuses it and so you end up with a butchered SDP.

I assume that on a SIP level, the T.38 re-invite is actually a re-invite and so is an offer. Use `rtpengine_offer` for that. If it's actually an answer on a SIP level too, then I'd say the SIP signalling is wrong, because I'm pretty sure you're not supposed to answer with T.38 to a non-T.38 offer.

**Comment by jpyle490** [comment]
Now that you've mentioned this I've watched the control traffic between OpenSIPS and rtpengine, and I see what you mean; it's calling ```answer``` on the offer.  I've verified I'm calling ```rtpengine_offer()``` from the script when I see the ```answer``` command to rtpengine, so...this is very odd.  I'll continue to dig.

**Comment by jpyle490** [comment]
Alright, I think I've got this.  The OpenSIPS 3.1.x documentation for the rtpengine module says ```force-answer``` will "only rewrite SDP when corresponding session already exists in the RTP proxy."  That to me meant it will only update an existing media session rather than create a new one, which is perfect for a re-INVITE.  It appears what it really means is that it does what the name implies, it forces the ```answer``` command to rtpengine, which is not helpful on an offer.

I removed ```force-answer``` from the rtpengine_offer() parameter string and the SDPs are processing correctly now.
